compiler/rustc_metadata/src/creader.rs

```diff
@@ -769,27 +769,31 @@ impl<'a> CrateLoader<'a> {
     }
 
     fn inject_profiler_runtime(&mut self, krate: &ast::Crate) {
-        if (self.sess.instrument_coverage()
+        if self.sess.instrument_coverage()
             || self.sess.opts.debugging_opts.profile
-            || self.sess.opts.cg.profile_generate.enabled())
-            && !self.sess.opts.debugging_opts.no_profiler_runtime
+            || self.sess.opts.cg.profile_generate.enabled()
         {
-            info!("loading profiler");
-
-            if self.sess.contains_name(&krate.attrs, sym::no_core) {
-                self.sess.err(
-                    "`profiler_builtins` crate (required by compiler options) \
-                               is not compatible with crate attribute `#![no_core]`",
-                );
-            }
+            if let Some(name) =
+                self.sess.opts.debugging_opts.profiler_runtime.as_deref().map(Symbol::intern)
+            {
+                info!("loading profiler");
+
+                if name == sym::profiler_builtins
+                    && self.sess.contains_name(&krate.attrs, sym::no_core)
+                {
+                    self.sess.err(
+                        "`profiler_builtins` crate (required by compiler options) \
+                                is not compatible with crate attribute `#![no_core]`",
+                    );
+                }
 
```

Comment:
The nesting here is starting to get a messy. Maybe just early return for the first if and if `profiler_runtime` is `None`?

Reply:
Thanks, have done!